ols/src/cache/redis_cache.py

```diff
@@ -70,14 +83,30 @@ def insert_or_append(self, user_id: str, conversation_id: str, value: str) -> No
             value: The value to set.
 
         Raises:
-            OutOfMemoryError: If item is evicted when Redis allocated
-                memory is higher than maxmemory.
+            RedisConnectionError: If item is unable to update after REDIS_MAX_RETRY.
         """
         key = super().construct_key(user_id, conversation_id)
-
         old_value = self.get(user_id, conversation_id)
-        with self._lock:
-            if old_value:
-                self.redis_client.set(key, old_value + "\n" + value)
-            else:
-                self.redis_client.set(key, value)
+        retry_count = 0
+        while retry_count < constants.REDIS_MAX_RETRY:
+            logger.debug("Updating redis cache ")
+            with self._lock:
+                if old_value:
+                    old_value.append(value)
+                    try:
+                        self.redis_client.set(key, old_value)
+                        break
+                    except Exception as redis_err:
+                        logger.info(f"Exception during updating the cache: {redis_err}")
+                        retry_count += 1
+                else:
+                    values = [value]
+                    try:
+                        self.redis_client.set(key, values)
+                        break
+                    except Exception as redis_err:
+                        logger.info(f"Exception during creating the cache: {redis_err}")
+                        retry_count += 1
+
+        if retry_count == constants.REDIS_MAX_RETRY:
+            raise RedisConnectionError("Updating cache failed after max retries")
```

Comment:
The intent is to use `RedisConnectionError` with different messages? Otherwise it should be something like `RedisMaxConnectionError`.

Reply:
IMHO if you `return` instead o `break` in the `while` loop, there's no nerd to check `retry_count` -> would be possible to simply raise an exception

Reply:
btw that `RedisMaxConnectionError` should be `RedisMaxRetryError`... my bad